### PR TITLE
Screen split

### PR DESCRIPTION
**Pull request by mhojune**
Split game screen

1. Made new class : Screen2P, GameScreen2P
2. Add some code : Core, DrawManager, Frame

### CODE REVIEW

File: src/engine/Core.java [hunk 1]
```diff
@@ -26,6 +22,8 @@ public final class Core {
 	private static final int WIDTH = 448;
 	/** Height of current screen. */
 	private static final int HEIGHT = 520;
+	/** Width of 2p screen*/
+	private static final int WIDTH2p = WIDTH*2;
```
Comment (logpacket) [inline src/engine/Core.java]: FOLLOW JAVA NAMING CONVENTION.

```java
private static final int WIDTH_2P = WIDTH*2;
```

File: src/engine/Core.java [hunk 2]
```diff
@@ -112,16 +115,16 @@ public static void main(final String[] args) {
 		gameSettings.add(SETTINGS_LEVEL_6);
 		gameSettings.add(SETTINGS_LEVEL_7);
 		
-		GameState gameState;
+		GameState gameState;		//gamestate 인스턴스 생성
```
Comment (logpacket) [inline src/engine/Core.java]: Don't use 한글.
Please write by english. Because, this course is for english and attend lot of foreigners.

Comment (logpacket) [reply src/engine/Core.java]: And this comment is unnecessary

File: src/engine/Core.java [hunk 3]
```diff
@@ -112,16 +115,16 @@ public static void main(final String[] args) {
...
 
-		int returnCode = 1;
+		int returnCode = 4;
 		do {
-			gameState = new GameState(1, 0, MAX_LIVES, 0, 0);
+			gameState = new GameState(1, 0, MAX_LIVES, 0, 0);	//gamestate에 값 할당 - 게임 상태 조정
```
Comment (logpacket) [inline src/engine/Core.java]: Unnecessary comment

File: src/engine/Core.java [hunk 4]
```diff
@@ -112,16 +115,16 @@ public static void main(final String[] args) {
...
 
 			switch (returnCode) {
 			case 1:
 				// Main menu.
-				currentScreen = new TitleScreen(width, height, FPS);
+				currentScreen = new TitleScreen(width, height, FPS);	//현재 스크린을 TitleScreen으로 바꾸기
```
Comment (logpacket) [inline src/engine/Core.java]: Unnecessary comment

File: src/engine/Core.java [hunk 5]
```diff
@@ -133,15 +136,15 @@ public static void main(final String[] args) {
 					// One extra live every few levels.
 					boolean bonusLife = gameState.getLevel()
 							% EXTRA_LIFE_FRECUENCY == 0
-							&& gameState.getLivesRemaining() < MAX_LIVES;
+							&& gameState.getLivesRemaining() < MAX_LIVES; // 스테이지 넘어갈 때 일정 주기로 MAX_LIVES 보다 남은 체력이 적다면 체력 회복 (기본값 3)
```
Comment (logpacket) [inline src/engine/Core.java]: Unnecessary comment

File: src/engine/Core.java [hunk 6]
```diff
@@ -133,15 +136,15 @@ public static void main(final String[] args) {
...
 					
-					currentScreen = new GameScreen(gameState,
+					currentScreen = new GameScreen(gameState,				//게임스크린 인스턴스 생성
```
Comment (logpacket) [inline src/engine/Core.java]: Unnecessary comment

File: src/engine/Core.java [hunk 7]
```diff
@@ -133,15 +136,15 @@ public static void main(final String[] args) {
...
 							gameSettings.get(gameState.getLevel() - 1),
 							bonusLife, width, height, FPS);
 					LOGGER.info("Starting " + WIDTH + "x" + HEIGHT
 							+ " game screen at " + FPS + " fps.");
-					frame.setScreen(currentScreen);
-					LOGGER.info("Closing game screen.");
+					frame.setScreen(currentScreen);							//현재 스크린(게임스크린) 띄우기, 여기서 대부분 처리(setscreen에는 run이 있음)
```
Comment (logpacket) [inline src/engine/Core.java]: Unnecessary comment

File: src/engine/Core.java [hunk 8]
```diff
@@ -133,15 +136,15 @@ public static void main(final String[] args) {
...
+					LOGGER.info("Closing game screen.");				//끝
```
Comment (logpacket) [inline src/engine/Core.java]: Unnecessary comment

File: src/engine/Core.java [hunk 9]
```diff
@@ -172,6 +175,45 @@ public static void main(final String[] args) {
 				returnCode = frame.setScreen(currentScreen);
 				LOGGER.info("Closing high score screen.");
 				break;
+			case 4:
+				DrawManager.getInstance().setFrame(frame2p);		//set 2p frame
```
Comment (logpacket) [inline src/engine/Core.java]: Unnecessary comment

File: src/engine/Core.java [hunk 10]
```diff
@@ -172,6 +175,45 @@ public static void main(final String[] args) {
...
+				//twoplayer
+				do {
+					// One extra live every few levels.
+					boolean bonusLife = gameState.getLevel()
+							% EXTRA_LIFE_FRECUENCY == 0
+							&& gameState.getLivesRemaining() < MAX_LIVES; // 스테이지 넘어갈 때 일정 주기로 MAX_LIVES 보다 남은 체력이 적다면 체력 회복 (기본값 3)
```
Comment (logpacket) [inline src/engine/Core.java]: Unnecessary comment

File: src/engine/Core.java [hunk 11]
```diff
@@ -172,6 +175,45 @@ public static void main(final String[] args) {
...
+
+					currentScreen = new TwoPlayerGameScreenDongbin(gameState,				//게임스크린 인스턴스 생성
```
Comment (logpacket) [inline src/engine/Core.java]: Unnecessary comment

File: src/engine/Core.java [hunk 12]
```diff
@@ -172,6 +175,45 @@ public static void main(final String[] args) {
...
+							gameSettings.get(gameState.getLevel() - 1),
+							bonusLife, width2p, height, FPS);					//create screen sized by 2pframe's width
+					LOGGER.info("Two player starting " + WIDTH + "x" + HEIGHT
+							+ " game screen at " + FPS + " fps.");
+					frame.setScreen(currentScreen);							//현재 스크린(게임스크린) 띄우기, 여기서 대부분 처리(setscreen에는 run이 있음)
```
Comment (logpacket) [inline src/engine/Core.java]: Unnecessary comment

File: src/engine/Core.java [hunk 13]
```diff
@@ -172,6 +175,45 @@ public static void main(final String[] args) {
...
+					LOGGER.info("Closing game screen.");				//끝
```
Comment (logpacket) [inline src/engine/Core.java]: Unnecessary comment

**Review by logpacket — COMMENTED** [review]
There are too many unnecessary comments.
Please rebase on master and remove that comments.

**Review by logpacket — CHANGES_REQUESTED** [review]
Conflict not resolved. Please check one more

File: src/engine/Core.java [hunk 14]
```diff
@@ -196,6 +202,46 @@ public static void main(final String[] args) {
 				returnCode = frame.setScreen(currentScreen);
 				LOGGER.info("Closing game setting screen.");
 
+			case 7:
+        //Twoplayer
```
Comment (logpacket) [inline src/engine/Core.java]: Please align indent

File: src/engine/Core.java [hunk 15]
```diff
@@ -135,6 +142,7 @@ public static void main(final String[] args) {
 					boolean bonusLife = gameState.getLevel()
 							% EXTRA_LIFE_FRECUENCY == 0
 							&& gameState.getLivesRemaining() < MAX_LIVES;
+
```
Comment (logpacket) [inline src/engine/Core.java]: Unnecessary change

File: src/engine/Core.java [hunk 16]
```diff
@@ -165,7 +173,6 @@ public static void main(final String[] args) {
 				returnCode = frame.setScreen(currentScreen);
 				LOGGER.info("Closing score screen.");
 				break;
-
```
Comment (logpacket) [inline src/engine/Core.java:168]: Unnecessary change

File: src/engine/Core.java [hunk 17]
```diff
@@ -201,7 +208,45 @@ public static void main(final String[] args) {
 						+ " game setting screen at " + FPS + " fps.");
 				returnCode = frame.setScreen(currentScreen);
 				LOGGER.info("Closing game setting screen.");
+			case 7:
+				DrawManager.getInstance().setFrame(frame2p);
+				//twoplayer
```
Comment (logpacket) [inline src/engine/Core.java]: If you want to describe this case, Please follow above comments.
Try referring this.
```java
case 7:
    //Two player
```

File: src/engine/Core.java [hunk 18]
```diff
@@ -201,7 +208,45 @@ public static void main(final String[] args) {
...
+				do {
+					// One extra live every few levels.
+					boolean bonusLife = gameState.getLevel()
```
Comment (logpacket) [inline src/engine/Core.java]: I think this logic should be used to control `GameScreen2p`, meaning that this logic should be applied to `Screen2P` for each player.
내가 생각하기에 이 로직은 `GameScreen2p`을 제어하는데 쓰여야 할것 같습니다. 즉 이 로직은 `Screen2P`에 각각의 플레이어에 따라 적용되어야 합니다.

File: src/screen/Screen2P.java [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+    public Screen2P(final GameState gameState,
+                    final GameSettings gameSettings, final boolean bonusLife,
+                    final int width, final int height, final int fps) {
+        super(width, height, fps);
+
+        player1Screen = new GameScreen2P(gameState, gameSettings, bonusLife, width /2 , height, fps, 1
+        );
+        player2Screen = new GameScreen2P(gameState, gameSettings, bonusLife, width / 2, height, fps, 2
+        );
+        executor = Executors.newFixedThreadPool(2);
```
Comment (logpacket) [inline src/screen/Screen2P.java]: You should create 2 instance of `GameState`.
`GameState` must be changed by each player.
Please check game play.
GameState` 인스턴스를 2개 생성해야 합니다.
게임 스테이트`는 각 플레이어가 변경해야 합니다.
게임 플레이를 확인해 주세요.

File: src/screen/Screen2P.java [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+
+        this.gameSettings = gameSettings;
+    }
+
+    public void show() throws Exception {
+        Future<Integer> player1Result = executor.submit(player1Screen);  // Starting the first game screen
+        Future<Integer> player2Result = executor.submit(player2Screen);  // Starting the second game screen
```
Comment (logpacket) [inline src/screen/Screen2P.java]: Remove unnecessary comments

File: src/screen/Screen2P.java [hunk 21]
```diff
@@ -0,0 +1,113 @@
...
+
+        // Update both screen
+        player1Screen.update();
+        player2Screen.update();
+
+    }
+
+    /**
+     * Draws the elements associated with the screen.
+     */
```
Comment (logpacket) [inline src/screen/Screen2P.java]: Remove this comments

File: src/engine/Core.java [hunk 22]
```diff
@@ -201,7 +208,45 @@ public static void main(final String[] args) {
...
+					gameState = new GameState(gameState.getLevel() + 1,
+							gameState.getScore(),
+							gameState.getLivesRemaining(),
+							gameState.getBulletsShot(),
+							gameState.getShipsDestroyed());
+
+				} while (gameState.getLivesRemaining() > 0
+						&& gameState.getLevel() <= NUM_LEVELS);
+
+				LOGGER.info("Starting " + WIDTH + "x" + HEIGHT
```
Comment (logpacket) [inline src/engine/Core.java:218]: It should be logging by each player.

File: src/engine/Frame.java [hunk 23]
```diff
@@ -32,7 +32,7 @@ public class Frame extends JFrame {
 	 */
 	public Frame(final int width, final int height) {
 		setSize(width, height);
-		setResizable(false);
```
Comment (logpacket) [inline src/engine/Frame.java:35]: I think this change is unnecessary.
Because, you created one more frame for 2p.

File: src/screen/GameScreen2P.java [hunk 24]
```diff
@@ -0,0 +1,350 @@
...
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+/**
+ * Implements the game screen, where the action happens.
+ *
+ * @author <a href="mailto:RobertoIA1987@gmail.com">Roberto Izquierdo Amo</a>
+ *
+ */
+public class GameScreen2P extends Screen implements Callable<Integer> {
```
Comment (logpacket) [inline src/screen/GameScreen2P.java]: I think, you can just extends `GameScreen` and override several methods. Please try my opinion

File: src/screen/Screen2P.java (hunk 21 above)
Comment (logpacket) [reply src/screen/Screen2P.java]: And I think you missed draw methods

**Review by logpacket — CHANGES_REQUESTED** [review]
Please check my opinion, and attach play screen shot to PR

Please remove unnecessary comments in your work. 
By useless comments, we mean
- Comments on existing code <-( These are the main culprits of merge conflicts).
- Comments next to code (especially method calls) that are not part of the method or member.
- Do not write comments in Korean, this is an English-only class and there are many English speakers in the class.

File: src/screen/Screen2P.java (hunk 21 above)
Comment (mhojune) [reply src/screen/Screen2P.java]: When player(n)Screen.update() is run, update run draw() in GameScreen2P. So i think draw methods in Screen2P is useless. What should i do?

Comment (logpacket) [reply src/screen/Screen2P.java]: If this screen is fine, just remove this comment
